pkg/proxy/endpoints.go

```diff
@@ -109,12 +113,19 @@ func (info *BaseEndpointInfo) GetZoneHints() sets.Set[string] {
 
 // IP returns just the IP part of the endpoint, it's a part of proxy.Endpoint interface.
 func (info *BaseEndpointInfo) IP() string {
-	return proxyutil.IPPart(info.Endpoint)
+	if info.ip == "" {
+		info.ip = proxyutil.IPPart(info.Endpoint)
```

Comment:
We always have the IP separate from the port at construct time (`newBaseEndpointInfo`), so we should just store it then, rather than parsing it back out here

Reply:
Because the `endpoints_test.go` file uses the `BaseEndpointInfo` structure directly and does not use the `newBaseEndpointInfo` method, it needs to be parsed once when the ip and port do not exist. This is a compatible approach, and there is a lot to change if you want to change `endpoints_test.go`. Of course, if you think this is recommended, I can make changes for that as well.

Reply:
yes, we should not make the code less efficient just to simplify the unit tests